trustfall/tests/helper_macros.rs

```diff
@@ -0,0 +1,130 @@
+// Do NOT import FieldValue, we want to check that accessor macros does what
+// we want
+
+struct Adapter;
+
+#[derive(Debug, Clone, trustfall::provider::TrustfallEnumVertex)]
+enum V {
+    Variant(Value),
+}
+
+#[derive(Debug, Clone)]
+struct Value {
+    name: String,
+}
+impl Value {
+    fn initial(&self) -> String {
+        self.name
+            .chars()
+            .next()
+            .map(String::from)
+            .unwrap_or_default()
+    }
+}
+
+impl trustfall::provider::BasicAdapter<'static> for Adapter {
+    type Vertex = V;
+    fn resolve_starting_vertices(
+        &mut self,
+        _edge_name: &str,
+        _parameters: &trustfall::provider::EdgeParameters,
+    ) -> trustfall::provider::VertexIterator<'static, Self::Vertex> {
+        Box::new(std::iter::once(V::Variant(Value {
+            name: String::from("Berit"),
+        })))
+    }
+    fn resolve_property(
+        &mut self,
+        contexts: trustfall::provider::ContextIterator<'static, Self::Vertex>,
+        type_name: &str,
+        property_name: &str,
+    ) -> trustfall::provider::ContextOutcomeIterator<'static, Self::Vertex, trustfall::FieldValue>
+    {
+        match (type_name, property_name) {
+            ("Vertex", "name") => trustfall::provider::resolve_property_with(
+                contexts,
+                trustfall::provider::field_property!(as_variant, name),
+            ),
+            ("Vertex", "initial") => trustfall::provider::resolve_property_with(
+                contexts,
+                trustfall::provider::accessor_property!(as_variant, initial),
+            ),
+            (t, p) => unreachable!("tried to resolve ({t}, {p})"),
+        }
+    }
+
+    fn resolve_neighbors(
+        &mut self,
+        _contexts: trustfall::provider::ContextIterator<'static, Self::Vertex>,
+        _type_name: &str,
+        _edge_name: &str,
+        _parameters: &trustfall::provider::EdgeParameters,
+    ) -> trustfall::provider::ContextOutcomeIterator<
+        'static,
+        Self::Vertex,
+        trustfall::provider::VertexIterator<'static, Self::Vertex>,
+    > {
+        todo!("schema should not contain neighbors: Berit likes it that way")
```

Comment:
haha who is Berit? 😂

Reply:
Extremely specific south Swedish internet sensation :smiley:

Reply:
Amazing!